### PR TITLE
Connected the register business page

### LINKED ISSUE #66: Connect List your business page with Backend
Use redux and axios.

### PR DESCRIPTION
**Pull request by ritwizsinha**
Fixes #66 

Changes: Made change in the authReduscer ,authAction

Surge Deployment Link: https://pr-${#79}-sahayaak.surge.sh

### CODE REVIEW

File: src/redux/actions/authActions.js [hunk 1]
```diff
@@ -1,89 +1,143 @@
...
+      );
+      dispatch({ type: LOGIN_FAIL });
+    });
+};
+export const registerBusiness = ({
+  businessname,
+  description,
+  residence,
+  phone
+}) => (dispatch, getState) => {
```
Comment (harsh-jindal) [inline src/redux/actions/authActions.js]: Put this in one line. Like this `export const registerBusiness = ({businessname,description,residence,phone}) => (dispatch, getState) => {`

File: src/redux/actions/authActions.js [hunk 2]
```diff
@@ -1,89 +1,143 @@
...
+  const detail = {
+    description: description,
```
Comment (harsh-jindal) [inline src/redux/actions/authActions.js]: when the name and value is same you can just write like it as 
`const detail = {
description,
etc..
}
`

File: src/redux/actions/authActions.js [hunk 3]
```diff
@@ -1,89 +1,143 @@
...
+    location: residence,
+    contact: phone
+  };
+  const name = businessname;
```
Comment (harsh-jindal) [inline src/redux/actions/authActions.js]: Remove this. 
Directly use businessname in the body object.

File: src/redux/actions/authActions.js [hunk 4]
```diff
@@ -1,89 +1,143 @@
...
+  const body = JSON.stringify({ name, detail });
+  const config = tokenConfig(getState);
+  axios
+    .post("/services", body,config)
+    .then(res =>
+      dispatch({
+        type: REGISTER_BUSINESS_SUCCESS,
+        body: res.data
```
Comment (harsh-jindal) [inline src/redux/actions/authActions.js]: Rename body to payload

**Review by harsh-jindal — CHANGES_REQUESTED** [review]

**Comment by harsh-jindal** [comment]
@ritwizsinha Make the above changes and create a separate action and reducer for vendor.
And squash your commits

File: src/redux/actions/authActions.js (hunk 4 above)
Comment (ritwizsinha) [reply src/redux/actions/authActions.js]: done

File: src/redux/actions/authActions.js (hunk 1 above)
Comment (ritwizsinha) [reply src/redux/actions/authActions.js]: done

File: src/redux/reducers/vendorReducer.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  case REGISTER_BUSINESS_SUCCESS: 
+  return{
+    ...state,
+    msg: action.payload
+  }
+  default:
+  return state;
+
+
```
Comment (harsh-jindal) [inline src/redux/reducers/vendorReducer.js]: Please don't leave empty lines in between on code.

Comment (ritwizsinha) [reply src/redux/reducers/vendorReducer.js]: done

File: src/redux/reducers/vendorReducer.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import {REGISTER_BUSINESS_FAIL,REGISTER_BUSINESS_SUCCESS} from '../actions/type';
+
+const initialState = {
```
Comment (harsh-jindal) [inline src/redux/reducers/vendorReducer.js:3]: @ritwizsinha You don't need all this inside the initial state. Just have an msg:''.

File: src/redux/reducers/vendorReducer.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  user: null,
+  openloginModal: false,
+  openregisterModal: false
+}
+
+export default function(state= initialState,action){
+  switch(action.type){
+  case REGISTER_BUSINESS_FAIL:
+    localStorage.removeItem('token');
+    localStorage.removeItem('user');
```
Comment (harsh-jindal) [inline src/redux/reducers/vendorReducer.js]: Don't log the user out on register business fail just set the msg and status which is received from the server.
If a status of 200 is not received the display a warning alert on the list your business page else a success alert.

**Review by harsh-jindal — CHANGES_REQUESTED** [review]

File: src/redux/reducers/vendorReducer.js (hunk 6 above)
Comment (ritwizsinha) [reply src/redux/reducers/vendorReducer.js:3]: done

File: src/redux/reducers/vendorReducer.js (hunk 7 above)
Comment (ritwizsinha) [reply src/redux/reducers/vendorReducer.js]: done

**Comment by ritwizsinha** [comment]
Made then changes
Screenshot from 2019-05-22 16-41-51

File: src/components/VendorSalesPage/VendorSalesPage.js [hunk 8]
```diff
@@ -32,49 +49,16 @@ class VendorSalesPage extends Component {
     const value = e.target.value;
     this.setState({ confirmDirty: this.state.confirmDirty || !!value });
```
Comment (harsh-jindal) [inline src/components/VendorSalesPage/VendorSalesPage.js]: What is !!value ?

Comment (harsh-jindal) [inline src/components/VendorSalesPage/VendorSalesPage.js:50]: What is !!value ?

**Review by harsh-jindal — APPROVED** [review]

**Pull request merged**